cdc/sink/ddlsink/mq/ddlproducer/pulsar_ddl_producer.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddlproducer
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/model"
+	// pulsarMetric "github.com/pingcap/tiflow/cdc/sink/metrics/mq/pulsar"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/sink/codec/common"
+	pulsarConfig "github.com/pingcap/tiflow/pkg/sink/pulsar"
+	"go.uber.org/zap"
+)
+
+// Assert DDLEventSink implementation
+var _ DDLProducer = (*pulsarProducers)(nil)
+
+// pulsarProducers is a producer for pulsar
+type pulsarProducers struct {
+	client           pulsar.Client
+	pConfig          *pulsarConfig.Config
+	defaultTopicName string
+	// support multiple topics
+	producers      map[string]pulsar.Producer
+	producersMutex sync.RWMutex
+	id             model.ChangeFeedID
+}
+
+// SyncBroadcastMessage pulsar consume all partitions
+func (p *pulsarProducers) SyncBroadcastMessage(ctx context.Context, topic string,
+	totalPartitionsNum int32, message *common.Message,
+) error {
+	// call SyncSendMessage
+	// pulsar consumer all partitions
+	return p.SyncSendMessage(ctx, topic, totalPartitionsNum, message)
+}
+
+// SyncSendMessage sends a message
+// partitionNum is not used,pulsar consume all partitions
+func (p *pulsarProducers) SyncSendMessage(ctx context.Context, topic string,
+	partitionNum int32, message *common.Message,
+) error {
+	p.wrapperSchemaAndTopic(message)
+
+	// pulsarMetric.IncPublishedDDLEventCountMetric(topic, p.id.ID, message)
+	producer, err := p.GetProducerByTopic(topic)
+	if err != nil {
+		log.L().Error("ddl SyncSendMessage GetProducerByTopic fail", zap.Error(err))
```

Comment:
```suggestion
		log.Error("ddl SyncSendMessage GetProducerByTopic fail", zap.Error(err))  
```
Using log.xxx is enough.